eng/tools/mcp/azure-sdk-python-mcp/main.py

```diff
@@ -196,22 +196,28 @@ def tox_tool(package_path: str, environment: str, repo_path: str, tox_ini_path:
     return run_command(command, cwd=repo_path)
 
 @mcp.tool("init")
-def init_tool(tsp_config_url: str, repo_path: str) -> Dict[str, Any]:
-    """Initializes and generates a typespec client library directory given the url.
+def init_tool(tsp_config_path: str, repo_path: str, is_local: bool = False) -> Dict[str, Any]:
+    """Initializes a new typespec client library directory.
     
     Args:
-        tsp_config_url: The URL to the tspconfig.yaml file.
-        repo_path: The path to the repository root (i.e. ./azure-sdk-for-python/).
+        :param str tsp_config_path: Either the URL to the tspconfig.yaml file or the path to a local tspconfig.yaml file.
+        :param str repo_path: The path to the repository root (i.e. ./azure-sdk-for-python/).
+        :param bool is_local: Whether the tsp_config_path is a local file path (True) or a URL (False).
+
     Returns:
         A dictionary containing the result of the command.
     """
     try:
-        # Get updated URL with latest commit hash
-        updated_url = get_latest_commit(tsp_config_url)
+        config_path = tsp_config_path
+        
+        # If not a local path, get updated URL with latest commit hash
+        if not is_local and tsp_config_path.startswith("http"):
+            config_path = get_latest_commit(tsp_config_path)
```

Comment:
What if someone passed in a url with the commit already in it?

Reply:
the func should handle that